plugins/module_utils/azure_rm_common_ext.py

```diff
@@ -213,44 +257,61 @@ def default_compare(self, modifiers, new, old, path, result):
 
+    def update_managed_identity(self, curr_identity=None, new_identity=None, allow_identities_append=True):
+        curr_identity = curr_identity or dict()
+        # TODO need to remove self.module.params.get('identity', {})
+        # after changing all modules to provide the "new_identity" parameter
+        # curr_identity and new_identity need to be mandatory parameters
+        new_identity = new_identity or self.module.params.get('identity', {}) or dict()
+        curr_managed_type = curr_identity.get('type', 'None')
+        new_managed_type = new_identity.get('type', 'None')
         # If type set to None, and Resource has None, nothing to do
-        if 'None' in param_identity.get('type', 'None') and current_managed_type == 'None':
-            pass
+        if new_managed_type == 'None' and curr_managed_type == 'None':
+            return False, None
+
+        changed = False
         # If type set to None, and Resource has current identities, remove UserAssigned identities
-        elif param_identity.get('type', 'None') == 'None':
-            changed = True
+        # Or
         # If type in module args different from current type, update identities
-        elif current_managed_type != param_identity.get('type', 'None'):
+        if new_managed_type == 'None' or curr_managed_type != new_managed_type:
             changed = True
 
+        curr_user_assigned_identities = set(curr_identity.get('user_assigned_identities', {}).keys())
+        new_user_assigned_identities = set(new_identity.get('user_assigned_identities', {}).get('id', []))
+        result_user_assigned_identities = new_user_assigned_identities
+
         # If type in module args contains 'UserAssigned'
-        if 'UserAssigned' in param_identity.get('type', 'None'):
-            if param_identity.get('user_assigned_identities', {}).get('append', False) is True:
-                external_identities = param_identities.union(current_managed_identities)
-                if len(current_managed_identities) != len(external_identities):
-                    # update identities
-                    changed = True
-            # If new identities have to overwrite current identities
-            else:
-                # Check if module args identities are different as current ones
-                if current_managed_identities.difference(external_identities) != set():
-                    changed = True
+        if allow_identities_append and \
+           'UserAssigned' in new_managed_type and \
+           new_identity.get('user_assigned_identities', {}).get('append', True) is True:
+            result_user_assigned_identities = new_user_assigned_identities.union(curr_user_assigned_identities)
 
-        new_identity = self.managed_identity['identity'](type=param_identity.get('type'))
+        # Check if module args identities are different as current ones
+        if curr_user_assigned_identities.difference(result_user_assigned_identities) != set():
```

Comment:
Please make the following change:
```
-        if curr_user_assigned_identities.difference(result_user_assigned_identities) != set():
+        if result_user_assigned_identities.difference(curr_user_assigned_identities) != set():
```
The order matters:

```
>>> foo
{'one'}
>>> bar
{'one', 'two'}
>>> foo.difference(bar)
set()
>>> bar.difference(foo)
{'two'}
```

Reply:
Fixed, thanks